frame/evm/precompile/contracts/encoder/src/lib.rs

```diff
@@ -23,12 +23,22 @@ use core::marker::PhantomData;
 // --- crates.io ---
 use codec::Encode;
 use evm::{executor::PrecompileOutput, Context, ExitError, ExitSucceed};
+use sha3::Digest;
 // --- darwinia-network ---
-use darwinia_support::evm::SELECTOR;
+use darwinia_support::s2s::RelayMessageCaller;
 use dp_evm::Precompile;
 // --- paritytech ---
 use frame_support::dispatch::{Dispatchable, GetDispatchInfo, PostDispatchInfo};
+use sp_std::prelude::Vec;
 
+const PALLET_DIG_LEN: usize = 4;
+const METHOD_DIG_LEN: usize = 4;
+const ACTION_LEN: usize = PALLET_DIG_LEN + METHOD_DIG_LEN;
+const BURN_AND_REMOTE_UNLOCK_METHOD: &[u8] = b"burn_and_remote_unlock()";
```

Comment:
Are these brackets necessary?

Reply:
Yes this can be deleted if we define it the same as in the contract. Here we use this format to describe it as a function. In fact, we should also add the parameters, but for simplicity, we ignore them.

Reply:
Is the method selector from the contract, if so the parameters should not be omitted

Reply:
This selector is used to map the caller. And the method is not the real function of the contract.